packages/terra-disclosure-manager/docs/README.md

```diff
@@ -0,0 +1,209 @@
+
+Example:
+```javascript
+app.disclose({
+  preferredType: 'disclosure-type',
+  size: 'large',
+  content: {
+    key: 'my-disclosed-content',
+    content: <DisclosedContent />,
+  }
+}).then({ dismissDisclosure, afterDismiss } => {
+  this.myContentIsDisclosed = true;
+
+  // This function can be cached and triggered as needed.
+  this.myDismissFunction = dismissDisclosure;
+
+  afterDismiss.then(() => {
+    // The afterDismiss will resolve when the content is dismissed, allowing for
+    // cleanup of local state as needed.
+    this.myContentIsDisclosed = false;
+
+    // If the dismissDisclosure function was cached, make sure it gets cleaned up.
+    this.myDismissFunction = false;
+  })
+}).catch((e) => {
+  // If this function is executed, the disclosure has been denied. You can use this to
+  // clean up any state that may be reliant on the disclosure ocurring.
+})
+```
+
+|Function|Description|
+|---|---|
+|`disclose(Object)`|Allows a component to disclose a component. This will open the disclosure if nothing was previously disclosed, or it will replace everything in the current disclosure stack. If disclosure was successful, it returns a Promise resolved with an Object containing the `dismissDisclosure` function and the `afterDismiss` deferred Promise. If disclosure was denied, it returns a rejected Promise.|
+
+`disclose` Argument API:
+
+|Key|Value|
+|---|---|
+|`preferredType`|The String describing the preferred disclosure type. This will be used to match the disclosure request to an appropriate DisclosureManager.|
+|`size`|The String size desired for the disclosure. One of `tiny`, `small`, `medium`, `large`, `huge`, or `fullscreen`. The functional implementation of this size is determined by the rendering component.|
+|`content`| An Object containing items describing the component to be disclosed.|
+
+`content` Object API:
+
+|Key|Value|
+|---|---|
+|key|A String key uniquely identifying the component to the DisclosureManager. This key will be automatically added to the component when rendered.|
+|component|A React element that will be disclosed.|
+
+#### Disclosure Content
+
+The AppDelegate instances provided to the disclosure components are a little more complicated. In addition to a `disclose` function (with all the bells and whistles described in the above section), a number of other functions are exposed to manage various segments of the disclosure state. The included functions are:
+
+|Function|Description|
+|---|---|
+|`disclose(Object)`|Allows a component to disclose another component on top of itself. See above for argument API.|
```

Comment:
Proposed re-wording?:
>Allows a component to disclose another component. See above for argument API. 
The disclosure manager implementation will determine how the next object will be disclosed.